lib/src/local_working_copy.rs

```diff
@@ -763,8 +765,34 @@ fn mtime_from_metadata(metadata: &Metadata) -> MillisSinceEpoch {
     )
 }
 
+/// On Windows, checks if a path is a reparse point (junction or symlink).
+/// Junctions have is_dir()=true but is_symlink()=false, so we need this
+/// separate check to avoid recursing into them and hitting "Access denied".
+#[cfg(windows)]
+fn is_reparse_point(metadata: &Metadata) -> bool {
+    const FILE_ATTRIBUTE_REPARSE_POINT: u32 = 0x400;
+    (metadata.file_attributes() & FILE_ATTRIBUTE_REPARSE_POINT) != 0
+}
```

Comment:
nit: can you inline this function in `is_normal_dir()`? There are no other callers.